front_end/panels/ai_chat/ui/settings/components/ModelSelectorFactory.ts

```
@@ -0,0 +1,121 @@
+ */
+export function createModelSelector(
+  container: HTMLElement,
+  labelText: string,
+  description: string,
+  selectorType: string,
+  modelOptions: ModelOption[],
+  selectedModel: string,
+  defaultOptionText: string,
+  onFocus?: () => void,
+): HTMLElement {
+  const modelContainer = document.createElement('div');
```

ESLint

[error] Prefer template literals over imperative DOM API calls

(rulesdir/no-imperative-dom-api)